Bottom: 7169246c1dcdabd9c359682283ebe77b7f4baf40
Top:    4dd125a021e086b82e6d26117b6cb498fcc3b9e8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 19:06:05 +0000

Change unmodifed parameters to GetValue to const char *

Silences some compiler warnings in action.c


---

diff --git a/src/action.c b/src/action.c
index 8d5931a..b8707d1 100644
--- a/src/action.c
+++ b/src/action.c
@@ -7012,7 +7012,8 @@ static int
 parse_layout_attribute_units (char *name, int def)
 {
   const char *as, *units = NULL;
-  int n = 0, v, r;
+  int n = 0, v;
+  bool r;
 
   as = AttributeGet (PCB, name);
   if (!as)
@@ -7653,7 +7654,7 @@ ActionImport (int argc, char **argv, int x, int y)
 	}
       else if (ys)
 	{
-	  int r;
+	  bool r;
 	  x = GetValue (xs, units, &r);
 	  y = GetValue (ys, units, &r);
 	}
diff --git a/src/misc.c b/src/misc.c
index 76b05c0..b1d4304 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -108,7 +108,7 @@ static struct
  * bool variable absolute to false if it leads with a +/- character
  */
 float
-GetValue (char *val, char *units, bool * absolute)
+GetValue (const char *val, const char *units, bool * absolute)
 {
   double value;
   int n = -1;
diff --git a/src/misc.h b/src/misc.h
index bb88a21..82bd194 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -74,7 +74,7 @@ BoxTypePtr GetArcEnds (ArcTypePtr);
 void ChangeArcAngles (LayerTypePtr, ArcTypePtr, long int, long int);
 char *UniqueElementName (DataTypePtr, char *);
 void AttachForCopy (LocationType, LocationType);
-float GetValue (char *, char *, bool *);
+float GetValue (const char *, const char *, bool *);
 int FileExists (const char *);
 char *Concat (const char *, ...);	/* end with NULL */
